libyul/optimiser/Suite.cpp

```diff
@@ -101,6 +102,7 @@ void OptimiserSuite::run(
 		CommonSubexpressionEliminator{_dialect}(ast);
 
 		(FunctionGrouper{})(ast);
+		EquivalentFunctionCombiner::run(ast);
```

Comment:
I think it makes sense to do this right before inlining - otherwise the inliner might inline equivalent functions, that are used rarely/only once. This way we have one hit it seems.

Reply:
Funnily resulting in one *more* function in the end, since inlining is prevented.

Reply:
Which is good - actually I realized that function inlining is perhaps not one of the most efficient optimizer steps, because code is very expensive.

Reply:
Hm - do you think equivalent functions might have "diverged" later on or just to have them removed early if possible? Originally I thought equivalent functions would/should not diverge, but thinking about it some steps might depend on some ordering induced by identifiers... anyways, I'll just add it there - hopefully it's not *too* expensive given the heuristics.

Reply:
My reasoning was that if we do that early, subsequent steps will be cheaper because there are fewer functions. Oh we should also run the unused pruner then of course.

Reply:
Yes, when running the pruner as well it make more sense. I'm still a bit worried that the heuristic may not be good enough and consequently the step could be rather expensive and shouldn't be run too often (especially since ``codeSize`` ignores a lot now)... but if that's a problem we can just improve the heuristic - it shouldn't be too hard to replace ``codeSize`` by some FNV-style identifier-agnostic hash function on the code.

Reply:
Added both right after the grouper. And it cut quite a few of the suffices in half, so it may in fact be worth it :-).

Reply:
Let's see - I actually don't think it's that expensive. code size just performs a single traversal of the AST.